GAE/src/org/waterforpeople/mapping/app/web/rest/QuestionAnswerRestService.java

```diff
@@ -227,12 +228,20 @@
                     SurveyedLocaleDao slDao = new SurveyedLocaleDao();
                     List<SurveyalValue> svals = slDao.listSVByQuestionAndSurveyInstance(
                             surveyInstanceId, Long.parseLong(questionId));
+                    Long surveyedLocaleId = null;
                     if (svals != null && svals.size() > 0) {
                         SurveyalValue sval = svals.get(0);
                         sval.setStringValue(qa.getValue());
                         slDao.save(sval);
+                        // Populate locale id from the only entity containing this attribute
+                        surveyedLocaleId = sval.getSurveyedLocaleId();
                     }
-
+                    
+                    // Update datapoint names for this survey, if applies
+                    if (q.getLocaleNameFlag() && surveyedLocaleId != null) {
+                        DataProcessorRestServlet.scheduleDatapointNameAssembly(q.getSurveyId(), surveyedLocaleId);
```

Comment:
In order to avoid an extra query for checking the creation survey in the SurveyedLocale entity, I use the `getLocaleNameFag()` check as a preliminary check - this should only be set in registration surveys. If for whatever reason the question is mistakenly flagged, the task will find out this is not the registration form, and therefore just do nothing (name won't be updated).